api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -109,9 +117,17 @@ func (instance *OpenStackDataPlaneRole) InitConditions() {
 		condition.UnknownCondition(SetupReadyCondition, condition.InitReason, condition.InitReason),
 	)
 
-	// Only set Baremetal condition for baremetal provisioning
+	// Only set Baremetal related conditions if we have baremetal hosts included in the
+	// baremetalSetTemplate.
 	if len(instance.Spec.BaremetalSetTemplate.BaremetalHosts) > 0 {
-		cl = append(cl, *condition.UnknownCondition(RoleBareMetalProvisionReadyCondition, condition.InitReason, condition.InitReason))
+		bmConditionsList := []condition.Type{
+			NodeSetBareMetalProvisionReadyCondition,
+			NodeSetIPReservationReadyCondition,
```

Comment:
This is not correct. IPSet and DNS conditions have no dependency on baremetal provisioning.